### PR TITLE
[WIP] add github actions CI

### PR DESCRIPTION
**Pull request by tstenner**
The current CI landscape looks as follows:

- Appveyor: Qt preinstalled, Linux + Windows, easy to use
- Azure Pipelines: serious free processing power, unlikely to go away, all three platforms, vera complicated
- Travis CI: don't think it'll stay around in it's current form, no Windows, slow
- Github Actions: no separate login required, awesome integration, all three platforms, somewhat easy to use

I've grown a bit attached to Azure Pipelines and its 10 free concurrent CI jobs, but I wouldn't want to explain anyone how to use it. Despite the shortcomings, most labs still use Windows and OSX so that rules out Appveyor and Travis and only github actions is left.

### CODE REVIEW

**Comment by tstenner** [comment]
All three platforms are built, but the MacOS artifacts are huge (~90MB). Are there any switches to macdeployqt to leave out parts we don't need @cboulay?

File: .github/workflows/cppcmake.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        os:
+        - ubuntu-latest
+        - windows-latest
+        - macOS-latest
+      fail-fast: false
+    steps:
+    - uses: actions/checkout@v1
+    - name: CMake version
+      run: cmake --version
+    - name: download liblsl (Windows)
```
Comment (cboulay) [inline .github/workflows/cppcmake.yml]: should be `download liblsl (Ubuntu)`

**Comment by cboulay** [comment]
A couple points:

* If I use my local up-to-date branch of liblsl, build that anew, and use it for my `-DLSL_INSTALL_ROOT`, then macdeployqt is never run, and my .app file is missing the required libraries.

* If I download liblsl from its release page and use that, then macdeployqt is run, and I end up with an app that is 39.5 MB. Not great but not as bad as what the GitHub CI generates.

* If I add `-no-plugins` to the macdeployqt command, the resulting file is still 39.5 MB.

Here is the macdeployqt help text:

```
bin $./macdeployqt
Usage: macdeployqt app-bundle [options]

Options:
   -verbose=<0-3>     : 0 = no output, 1 = error/warning (default), 2 = normal, 3 = debug
   -no-plugins        : Skip plugin deployment
   -dmg               : Create a .dmg disk image
   -no-strip          : Don't run 'strip' on the binaries
   -use-debug-libs    : Deploy with debug versions of frameworks and plugins (implies -no-strip)
   -executable=<path> : Let the given executable use the deployed frameworks too
   -qmldir=<path>     : Scan for QML imports in the given path
   -qmlimport=<path>  : Add the given path to the QML module search locations
   -always-overwrite  : Copy files even if the target file exists
   -codesign=<ident>  : Run codesign with the given identity on all executables
   -appstore-compliant: Skip deployment of components that use private API
   -libpath=<path>    : Add the given path to the library search path
   -fs=<filesystem>   : Set the filesystem used for the .dmg disk image (defaults to HFS+)

macdeployqt takes an application bundle as input and makes it
self-contained by copying in the Qt frameworks and plugins that
the application uses.

Plugins related to a framework are copied in with the
framework. The accessibility, image formats, and text codec
plugins are always copied, unless "-no-plugins" is specified.

Qt plugins may use private API and will cause the app to be
rejected from the Mac App store. MacDeployQt will print a warning
when known incompatible plugins are deployed. Use -appstore-compliant 
to skip these plugins. Currently two SQL plugins are known to
be incompatible: qsqlodbc and qsqlpsql.
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,37 +1,32 @@
...
-
-#: This sets the minimum required CMake version.
-#: Here, [CMake 3.5](https://cmake.org/cmake/help/latest/release/3.5.html#modules)
-#: was chosen because it is the first version to support imported targets
-#: (e.g. `boost::thread`). Before setting this to a higher version, please check
-#: if it has arrived in [debian stable](https://packages.debian.org/stable/cmake).
-project(BestPracticesGUI
+# CMakeLists.txt contains the build configuration that can be used to build
+# your App without any IDE or to generate project files for your specific IDE.
+# https://packages.ubuntu.com/bionic/cmake
```
Comment (cboulay) [inline CMakeLists.txt]: This is a bit strange. Why link to the ubuntu package here?

File: mainwindow.cpp [hunk 3]
```diff
@@ -1,10 +1,9 @@
...
- * mainwindow.cpp contains the implementations of everything your window does. */
+/* mainwindow.cpp contains the implementations of everything your window does. */
 /*: The next two includes are our own headers that define the interfaces for
  * our window class and the recording device */
-#include "mainwindow.h"
-#include "reader.h"
-/*: `ui_mainwindow.h` is automatically generated from `mainwindow.ui`.
+#include "mainwindow.hpp"
+#include "reader.hpp"
+/*: `ui_mainwindow.hpp` is automatically generated from `mainwindow.ui`.
```
Comment (cboulay) [inline mainwindow.cpp]: .hpp extension is inconsistent with the `#include "ui_mainwindow.h"` below.

File: CMakeLists.txt (hunk 2 above)
Comment (tstenner) [reply CMakeLists.txt]: The main linux distributions are Debian and Ubuntu derivates so the required CMake version shouldn't be newer than the one on most developers' PCs.

File: mainwindow.cpp (hunk 3 above)
Comment (tstenner) [reply mainwindow.cpp]: Good catch, I've changed it back and will push it with the next batch of changes.

File: CMakeLists.txt (hunk 2 above)
Comment (cboulay) [reply CMakeLists.txt]: The links below make sense. The single link above this comment is the strange one.

Comment (tstenner) [reply CMakeLists.txt]: Right, also added to the fixup queue

**Comment by tstenner** [comment]
The install code in `LSLCMake.cmake` isn't quite right for MacOS, but our department Mac is currently in use.

The CI code here works, and as soon as I've fixed the MacOS install logic in liblsl it will include the bundled Qt libs.

**Pull request merged**